server/static/static.go

```diff
@@ -50,6 +50,49 @@ func (s *FilesServer) ServerFiles(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Strict-Transport-Security", "max-age=31536000")
 	}
 
-	// in my IDE (IntelliJ) the next line is red for some reason - but this is fine
-	ServeHTTP(w, r)
+	if r.URL.Path == "/" || !s.uiAssetExists(r.URL.Path) {
+		data, err := s.getIndexData()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
```

Comment:
This would be consistent with the previous behaviour because of the `!s.uiAssetExists`, just making that note because it isn't immediately obvious.

Reply:
@isubasinghe not sure about your mention **ByteReedSeeker**
but the !s.uiAssetExists match with previouse logic here https://github.com/argoproj/argo-workflows/blob/baef4856ff2603c76dbe277c825eaa3f9788fc91/server/static/static.go#L21-L24
So if static files does not exist, the index.html will be return